src/SourceBuild/content/repo-projects/Directory.Build.props

```diff
@@ -178,10 +183,21 @@
 
     <ArcadeSharedFrameworkSdkOverride Include="Microsoft.DotNet.SharedFramework.Sdk" Group="ARCADE_SHARED_FX_SDK" Version="$(arcadeOutputPackageVersion)" />
     <WindowsDesktopSdkOverride Include="Microsoft.Net.Sdk.WindowsDesktop" Group="WINDOWS_DESKTOP" Location="$(ToolsDir)EmptySdk" Condition="'$(DotNetBuildSourceOnly)' == 'true'" />
-    <NoTargetsSdkProject Include="$(SrcDir)source-build-reference-packages/**/microsoft.build.notargets.$(PrivateSourceBuiltNoTargetsSdkVersion).csproj"/>
-    <NoTargetsSdkOverride Include="@(NoTargetsSdkProject->'Microsoft.Build.NoTargets')" Group="NOTARGETS" Version="$(PrivateSourceBuiltNoTargetsSdkVersion)" />
-    <TraversalSdkProject Include="$(SrcDir)source-build-reference-packages/**/microsoft.build.traversal.$(PrivateSourceBuiltTraversalSdkVersion).csproj"/>
-    <TraversalSdkOverride Include="@(TraversalSdkProject->'Microsoft.Build.Traversal')" Group="TRAVERSAL" Version="$(PrivateSourceBuiltTraversalSdkVersion)" />
+
+    <!--
+      Add Microsoft.Build.NoTargets and Microsoft.Build.Traversal SDK overrides, for all repos,
+      except source-build-reference-packages repo, that builds these packages and utility projects.
+
+      We need to override all SDK packages that are used by repos, to prevent NuGet SDK Resolver from trying to
+      restore these packages from online feeds. In offline build we only want to use local source-build feeds.
+      Therefore, we create a copy of repo's NuGet.config file, modify it, and pass it into repo build using
+      RestoreConfigFile property. MSBuild uses NuGetSdkResolver to restore missing SDKs, which doesn't honor
+      RestoreConfigFile property and uses NuGet.config file from repo's root which has online feeds.
+    -->
+    <NoTargetsSdkOverride Include="Microsoft.Build.NoTargets" Group="NOTARGETS" Version="$(NoTargetsSdkVersion)" />
+    <TraversalSdkOverride Include="Microsoft.Build.Traversal" Group="TRAVERSAL" Version="$(TraversalSdkVersion)" />
+    <SourceBuiltSdkOverride Include="@(NoTargetsSdkOverride)" Condition="'$(RepositoryName)' != 'source-build-reference-packages' and '$(IsUtilityProject)' != 'true'" />
+    <SourceBuiltSdkOverride Include="@(TraversalSdkOverride)" Condition="'$(RepositoryName)' != 'source-build-reference-packages' and '$(IsUtilityProject)' != 'true'" />
```

Comment:
This works for source build but not for the non source-build case. You will need to update these items and set the `Location` metadata when not building from source and set to `$(NuGetPackageRoot).../$(NoTargetsSdkVersion)`.

Reply:
How about we don't even set these overrides in non-sourcebuild case (i.e. add `'$(DotNetBuildSourceOnly)' == 'true'`)?

Is there an offline scenario for non-sourcebuild, that would need these SDK overrides?

Reply:
Right, that's even better.